scripts/generate-renode-scripts.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+
+import csv
+import zlib
+import argparse
+
+def generate_ethmac(data):
+    base = get_entry_value(data, 'csr_base', 'ethmac')
+    buffer_base = get_entry_value(data, 'memory_region', 'ethmac')
+    buffer_size = get_entry_value(data, 'memory_region', 'ethmac', 1)
+    interrupt = get_entry_value(data, 'constant', 'ethmac_interrupt')
+
+    result = [
+        'ethmac: Network.LiteX_Ethernet @ {',
+        '       sysbus <{}, +0x100>;'.format(base),
+        '       sysbus new Bus.BusMultiRegistration {{ address: {}; size: {}; region: "buffer" }}'.format(buffer_base, buffer_size),
+        '    }'
+    ]
+
+    if interrupt is not None:
+        result.append('    -> cpu@{}'.format(interrupt))
+    return result
+
+def generate_peripheral(data, name, model, properties=None):
+    base = get_entry_value(data, 'csr_base', name)
+    interrupt = get_entry_value(data, 'constant', '{}_interrupt'.format(name))
+
+    result = ['{}: {} @ sysbus {}'.format(name, model, base)]
+    if properties is not None:
+        for pname, pvalue in properties.items():
+            if pvalue is not None:
+                result.append('    {}: {}'.format(pname, pvalue))
+    if interrupt is not None:
+        result.append('    -> cpu@{}'.format(interrupt))
+    return result
+
+def generate_memory_region(data, name):
+    base = get_entry_value(data, 'memory_region', name)
+    size = get_entry_value(data, 'memory_region', name, 1)
+
+    return [
+        '{}: Memory.MappedMemory @ sysbus {}'.format(name, base),
+        '    size: {}'.format(size)
+    ]
+
+def generate_silencer(data, name):
+    base = get_entry_value(data, 'csr_base', name)
+
+    return [
+        'sysbus:',
+        '    init add:',
+        '        SilenceRange <{} 0x200> # {}'.format(base, name)
+    ]
+
+def generate_flash_boot(data):
+    # TODO: this value is hardcoded, as I couldn't find it in the csv
+    base = 0x20228000
+    return [
+        'boot_flash: Memory.ArrayMemory @ sysbus {}'.format(base),
+        '    size: 0x00100000'
```

Comment:
The size / base comes from `mem.h` in `build/XXXX/software/include/generated/`
```c
#define SPIFLASH_BASE 0xa0000000
#define SPIFLASH_SIZE 0x01000000
```

Reply:
This is actually in the CSV file in `build/XXX/test/csr.csv` too;
```csv
memory_region,rom,0x00000000,32768,
memory_region,sram,0x10000000,32768,
memory_region,spiflash,0xa0000000,16777216,
memory_region,main_ram,0x40000000,268435456,
memory_region,ethmac,0xb0000000,8192,
```

Reply:
I was confused by the fact that in `include/generated/mem.h` `FLASH_BOOT_ADDRESS` is defined as `0x20228000`, but `spiflash` is located at `0xa0000000`.

Now I see in `targets/arty/base.py` that this address is a default shadow for `0x20000000`.
For Arty platform, the flash boot address is `spi_flash` base + gateware size (0x220000) + bios size (0x8000), so it all adds up.

It's still a bit problematic, as needed parameters (information about shadow address, gateware size or bios size) are not included in `csr.csv` so I don't see a way of calculating the final value based solely on this input.

Reply:
I looked a little in LiteX sources and it looks that the best option is to emit more information (i.e., `shadow_base` and `flash_boot_offset`) to `csr.csv` file.

As it will require a separate PR to LiteX that would block this one, I decided to temporarily remove the functionality of booting from flash (still leaving a netboot option available). I will add it with a separate PR, once the necessary code is in LiteX.